src/unleash.ts

```diff
@@ -34,18 +42,51 @@ export class Unleash extends EventEmitter {
             throw new Error('Unleash server URL missing');
         }
 
-        this.repository = new Repository(backupPath, url, refreshInterval);
+        if (!appName) {
+            throw new Error('Unleash client appName missing');
+        }
 
-        this.repository.on('error', (err) => {
-            this.emit('error', err);
-        });
+        if (!instanceId) {
+            instanceId = `generated-${Math.round(Math.random() * 1000000)}-${process.pid}`;
+        }
+
+        const requestId: string = `${appName}:${instanceId}`;
+        this.repository = new Repository(backupPath, url, requestId, refreshInterval);
 
         strategies = [new Strategy('default', true)].concat(strategies);
 
         this.repository.on('ready', () => {
             this.client = new Client(this.repository, strategies, errorHandler);
             this.emit('ready');
         });
+
+        this.repository.on('error', (err) => {
+            err.message = `Unleash Repository error: ${err.message}`;
+            this.emit('error', err);
+        });
+
+        this.repository.on('warn', (msg) => {
+            this.emit('warn', msg);
+        });
+
+        this.metrics = new Metrics({
```

Comment:
we might want a flag to disable metrics in case it prove to be some perf issues?